CodeConverter/CSharp/ExpressionNodeVisitor.cs

```diff
@@ -897,8 +897,10 @@ public override async Task<CSharpSyntaxNode> VisitBinaryExpression(VBasic.Syntax
                 return _visualBasicEqualityComparison.GetFullExpressionForVbObjectComparison(lhs, rhs, node.IsKind(VBasic.SyntaxKind.NotEqualsExpression));
         }
 
-        omitConversion |= lhsTypeInfo.Type != null && rhsTypeInfo.Type != null &&
-                          lhsTypeInfo.Type.IsEnumType() && SymbolEqualityComparer.IncludeNullability.Equals(lhsTypeInfo.Type, rhsTypeInfo.Type)
+        var lhsTypeIgnoringNullable = lhsTypeInfo.Type.GetNullableUnderlyingType() ?? lhsTypeInfo.Type;
+        var rhsTypeIgnoringNullable = rhsTypeInfo.Type.GetNullableUnderlyingType() ?? rhsTypeInfo.Type;
+        omitConversion |= lhsTypeIgnoringNullable != null && rhsTypeIgnoringNullable != null &&
+                          lhsTypeIgnoringNullable.IsEnumType() && SymbolEqualityComparer.Default.Equals(lhsTypeIgnoringNullable, rhsTypeIgnoringNullable)
```

Comment:
This was the Conversions.ToInteger issue. The ConvertedType says `int?` since that's presumably just always how enums are compared under the covers (hence this existing code), but because one side was nullable, this omitConversion option wasn't set